src/wasm_offset_converter.js

```diff
@@ -0,0 +1,82 @@
+function WasmOffsetConverter(wasmBytes) {
+  // This class parses a WASM binary file, and constructs a mapping from
+  // function indices to the start of their code in the binary file.
+  //
+  // The main purpose of this module is to enable the conversion of function
+  // index and offset from start of function to an offset into the WASM binary.
+  // This is needed to look up the WASM source map as well as generate
+  // consistent program counter representations given v8's non-standard
+  // WASM stack trace format.
+
+  var offset = 8;
+  var count;
+  var idx = 0;
+
+  this.map = {};
+
+  function unsignedLEB128() {
+    var result = 0;
+    var shift = 0;
+    do {
+      var byte = wasmBytes[offset++];
+      result += (byte & 0x7F) << shift;
+      shift += 7;
+    } while (byte & 0x80);
+    return result;
+  }
+
+  function skipLimits() {
+    switch (wasmBytes[offset++]) {
+      case 1: unsignedLEB128(); // has both initial and maximum, fall through
+      case 0: unsignedLEB128(); // just initial
+    }
+  }
+
+  while (offset < wasmBytes.length) {
+    var start = offset;
+    var type = wasmBytes[offset++];
+    var end = unsignedLEB128() + offset;
+    switch (type) {
+      case 2: // import section
```

Comment:
a comment explaining why we need to parse the import section could help, as it seems odd. i am guessing because function indexes are influenced by function imports?

Reply:
Yes, functions defined in the binary are numbered after all imports. I'll put this in a comment.